### PR TITLE
DEV-688 update oauth gems

### PR DESCRIPTION
**Pull request by jeremygottfried**
**Changes:** 
- Ruby 2.24 -> 2.7.5
- oauth2 -> 2.0.9
- omniauth -> 2.1.1
- omniauth-oauth2 -> 1.8.0
- `authorize_path` renamed to `authorize_url`
- `token_path` renamed to `token_url`
- `/` from relative urls automatically handled by oauth2

### CODE REVIEW

**Comment by theycallmeswift** [comment]
LGTM!

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.2.4
\ No newline at end of file
+2.7.5
```
Comment (theycallmeswift) [inline .ruby-version]: @jeremygottfried is there a reason we're keeping this at Ruby 2.x? If so, we should probably use Ruby 2.7.7, which is the latest release.

File: lib/omniauth-mlh/version.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 module OmniAuth
   module MLH
     MAJOR_VERSION = 0
```
Comment (theycallmeswift) [inline lib/omniauth-mlh/version.rb]: If we bump this to Ruby 3, we should consider a major bump to 1.0.0

File: omniauth-mlh.gemspec [hunk 3]
```diff
@@ -14,15 +14,16 @@ Gem::Specification.new do |spec|
   spec.homepage      = "http://github.com/mlh/omniauth-mlh"
   spec.license       = "MIT"
 
-  spec.required_ruby_version = '>= 2.2.4'
+  spec.required_ruby_version = '>= 2.7.5'
```
Comment (theycallmeswift) [inline omniauth-mlh.gemspec]: See comment above about Ruby version

**Review by theycallmeswift — COMMENTED** [review]
Only one set of comments about the Ruby version you selected. Otherwise LGTM.

Also, have you verified that the gem works as expected in your own dev environment or are you just relying on the tests to assume it works?

**Comment by theycallmeswift** [comment]
Here's a gist to test it locally. Confirmed it is working as expected with the current updates: https://gist.github.com/theycallmeswift/d8a8a22f95dd5d35f03661b19665e0d3

**Comment by jeremygottfried** [comment]
> Here's a gist to test it locally. Confirmed it is working as expected with the current updates: https://gist.github.com/theycallmeswift/d8a8a22f95dd5d35f03661b19665e0d3

@theycallmeswift I tried your gist with ruby 3.0.5, and it seems to return the correct redirect_uri in the url query params, but it says "not valid" on the view so I'm not sure which I'm supposed to pay attention to.
Screen Shot 2023-01-23 at 6 16 27 PM

**Comment by jeremygottfried** [comment]
@theycallmeswift ok looks like ruby v3.0.5 is working with the gist. I'm bumping gem version to 1.0.0

**Pull request closed without merging**